src/generic/clipboard/hooks/useCopyToClipboard.js

```diff
@@ -30,6 +34,22 @@ const useCopyToClipboard = (canEdit = true) => {
     setShowPasteUnit(!!isPasteableUnit);
   };
 
+  // Called on initial render to fetch and populate the initial clipboard data in redux state.
+  // Without this, the initial clipboard data redux state is always null.
+  useEffect(() => {
+    const fetchInitialClipboardData = async () => {
+      try {
+        const userClipboard = await getClipboard();
+        dispatch(updateClipboardData(userClipboard));
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(`Failed to fetch initial clipboard data: ${error}`);
+      }
+    };
+
+    fetchInitialClipboardData();
+  }, [dispatch]);
```

Comment:
As a general approach, I would prefer we replace the redux code with something that uses React Query, rather than finding a way to make the redux code work better. But this is fine for now.

See recently updated frontend recommendations: https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0067-bp-tools-and-technology.html#frontend-technology-selection

Reply:
Agreed, makes sense, I just wasn't sure if I wanted to do the full refactor to React Query as part of this task/PR, as opposed to a separate PR/task, since it's not specific to pasting in the library.